File: Aula_0503/Ex01.cpp
#include <iostream>

using namespace std;

class Aluno {
  private:
    string nome, matricula;
  public:
    void SetNome(string nome) {
      this->nome = nome;
    }
    void SetMatricula(string matricula) {
      this->matricula = matricula;
    }
    string ToString() { 
      return nome + " " + matricula ;}
};

int main() {
  Aluno a;
  a.SetNome("Gilbert");
  a.SetMatricula("1234");
  cout << a.ToString() << endl;
  Aluno b = a;
  cout << b.ToString() << endl;
  b.SetNome("Eduardo");
  cout << a.ToString() << endl;
  cout << b.ToString() << endl;

  Aluno *c = new Aluno();
  c->SetNome("Jorgiano");
  c->SetMatricula("4321");
  cout << c->ToString() << endl;
  delete c;
  c = &a;
  cout << c->ToString() << endl;

  cout << &a << endl;
  cout << &b << endl;
  cout << &c << endl;
  cout << c << endl;
}

File: Aula_0503/Ex01.cs
using System;

struct Aluno {
  public string nome, matricula;
  public override string ToString() {
    return $"{nome} - {matricula}";
  }
}

class Program {
  public static void Main() {
    Aluno a = new Aluno();
    a.nome = "Gilbert";
    a.matricula = "1234";
    Aluno b = new Aluno();
    b.nome = "Eduardo";
    b.matricula = "4321";
    Console.WriteLine(a);
    Console.WriteLine(b);
    a = b;
    a.nome = "Jorgiano";
    Console.WriteLine(a);
    Console.WriteLine(b);
  }
}
    